guides/micronaut-aws-parameter-store/micronaut-aws-parameter-store.adoc

```diff
@@ -0,0 +1,223 @@
+= @guideTitle@
+
+@guideIntro@
+
+Authors: @authors@
+
+Micronaut Version: @micronaut@
+
+== Getting Started
+
+In this guide, you are going to use AWS Parameter Store in Micronaut Application to drive configuration.
+
+include::{commondir}/common-requirements.adoc[]
+
+include::{commondir}/common-completesolution.adoc[]
+
+include::{commondir}/common-create-app.adoc[]
+
+=== Controller
+
+First, create an interface to encapsulate the Value-added Tax rate of a country.
+
+source:Vat[]
+
+Create a GET endpoint `/vat`. It returns the Value-added Tax rate configured in our app.
+
+source:VatController[]
+
+<1> The class is defined as a controller with the @Controller annotation.
+<2> Use constructor injection to inject a bean of type `Vat`
+<3> Since it does not return JSON, set the media type to `text/plain`
+<4> The @Get annotation is used to map the index method to all requests that use an HTTP GET to `/vat`
+
+=== Configuration
+
+source:VatConfiguration[]
+
+<1> The `@ConfigurationProperties` annotation takes the configuration prefix
+<2> By implementing the interface, we can inject a bean of type `Vat` in the previous Controller's constructor.
+
+=== Test
+
+Write a test which verifies a GET `/vat` endpoint returns the value set via configuration.
+
+test:VatControllerTest[]
+
+<1> Annotate the class with `@MicronautTest` so Micronaut will initialize the application context and the embedded server.
+<2> Annotate the class with `@Property` to supply configuration to the test.
+
+=== Dependency
+
+To use https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html[AWS Parameter Store]
+to drive configuration add the following dependency:
+
+dependency:micronaut-aws-parameter-store[groupId=io.micronaut.aws]
+
+=== Bootstrap.yml
+
+To enable distributed configuration a src/main/resources/bootstrap.yml configuration file must be created and configured to use Parameter Store:
+
+resource:bootstrap.yml[]
+
+`micronaut.application.name` name will be part of the parameter name in AWS Parameter Store.
+With `aws.client.system-manager.parameterstore.enabled` enable distributed configuration with AWS Parameter Store.
```

Comment:
I think this is missing here:
```
micronaut:
  config-client:
    enabled: true
```

Reply:
Good catch. The feature generates that. https://github.com/micronaut-projects/micronaut-starter/blob/2.4.x/starter-core/src/main/java/io/micronaut/starter/feature/awsparameterstore/AwsParameterStore.java#L49 but that meeting does not appear in our [configuration documentation](https://docs.micronaut.io/latest/guide/configurationreference.html#io.micronaut.runtime.ApplicationConfiguration). I am going to look in the javadoc to see if I find what it does.

Reply:
I've added a sentence to the guide about `micronaut.config-client.enabled`. Moreover, the javadoc seemed to be wrong as well. https://github.com/micronaut-projects/micronaut-core/pull/5178